### ISSUE TITLE
linter: `jest/max-expects` false positive (certainly caused by `mi.mocked`)

### ISSUE THREAD

**Issue by antoinezanardi** [oxc-project/oxc#15942]
### What version of Oxlint are you using?

 1.29.0

### What command did you run?

oxlint --type-aware -c configs/oxlint/oxlint.config.jsonc

### What does your `.oxlintrc.json` config file look like?

```jsonc
 "jest/max-expects": ["error",
    {
      "max": 1
    }
],
```

### What happened?

`jest/max-expects` rule is reporting a false positive. 

Consider the following code : 
```typescript
expect(vi.mocked(createQuestionThemeFromDocument)).toHaveBeenCalledExactlyOnceWith(foundQuestionThemeDocument);
```

Oxc reports :
```bash
× eslint-plugin-jest(max-expects): Enforces a maximum number assertion calls in a test body.
     ╭─[src/contexts/question/infrastructure/persistence/mongoose/question-theme/repository/question-theme.mongoose.repository.spec.ts:100:7]
  99 │ 
 100 │       expect(vi.mocked(createQuestionThemeFromDocument)).toHaveBeenCalledExactlyOnceWith(foundQuestionThemeDocument);
     ·       ──────
 101 │     });
     ╰────
```

IMHO, `vi.mocked` is resolved by the linter as an `expect` statement, thus reporting `2` expects.

If I put the `vi.mocked` in a variable outside of the test, it works as expected.

**Comment by antoinezanardi** [comment]
Can't reproduce the issue, it looks like it's fixed now ! Closing it.
